### DISCUSSION TITLE
How to Deploy Same Helm Chart, Multiple Times, Into Different Namespaces

### DISCUSSION THREAD

**Discussion by jseiser** [fluxcd/flux2#3139] [Q&A] [answered]
We are building out a small cluster for a dev team.  

Ive been working through this repo: https://github.com/fluxcd/flux2-kustomize-helm-example

The `infrastructure` part went fine.

Now instead of `apps` I need to create a way for each developer, to deploy/maintain their own version of the application they are working on.  

```
├── clusters
│   └── qa
│       ├── deploys.bak
│       ├── flux-system
│       │   ├── gotk-components.yaml
│       │   ├── gotk-sync.yaml
│       │   └── kustomization.yaml
│       └── infrastructure.yaml
├── deploys
│   ├── base
│   ├── dev1
│   ├── dev2
│   ├── dev3
│   └── staging
```

In `deploys/base` it would be great to specify a Namespace, a HelmRelease, and a Kubernetes Secret.  

Then in deploys/dev1 it would be great if we could include the base but have a way of overriding the `namespace` everything goes into.

So you would have  namespaces `app-dev1`, `app-dev2` etc.

This would allow us to only really have to override the ingress information, and the image tag for the app. 

Thanks for any information on this.

**Comment by Rohit27mgt** [comment]
I am doing the same but image pull secret causing error while secret are present in each namespace 

Can you resolve my problem or need more info to understand please let me know

**Reply by matheuscscp** [reply, marked as answer]
With flux-operator's ResourceSet you can achieve what you need:

https://fluxcd.control-plane.io/operator/resourcesets/app-definition/#multi-instance-example

If you add the following annotation to the Secret definition inside the ResourceSet resources you can copy it from some namespace:

```yaml
        annotations:
          fluxcd.controlplane.io/copyFrom: "<secret namespace>/<secret name>"
```

In this case you won't create the original Secret with ResourceSet but rather just copy it into the other namespaces.

Here are the docs explaining how to copy image pull secrets across namespaces: https://fluxcd.control-plane.io/operator/resourceset/#copying-data-from-existing-configmaps-and-secrets

**Reply by stefanprodan** [reply]
Here are the docs explaining how to copy image pull secrets across namespaces: https://fluxcd.control-plane.io/operator/resourceset/#copying-data-from-existing-configmaps-and-secrets

**Comment by Rohit27mgt** [comment]
But I am creating secret using a single cmd got image pull secret and I am successfully able to use in one namespace but getting imagepullbackoff on another ns

**Reply by matheuscscp** [reply]
Yes, with ResourceSet you can copy the secret from that namespace into the others

**Reply by matheuscscp** [reply]
I updated my answer above, check it out.
